ip-assignment-watchdog.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 Allterco Robotics EOOD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Shelly is a Trademark of Allterco Robotics
+
+// Shelly Script example: IP Assignment Watchdog
+//
+// This script monitors if the Shelly devices has a valid IP assignment from the DHCP of the router,
+// and if there is no valid IP for a few minutes - reboots the Shelly device. This is useful in scenarios
+// where the router's configuration has changed or the router was restarted, but the Shelly devices
+// keeps hanging on to the old connection for some reason. In essence, this is am automated
+// self-healing mechanism in case you have installed many Shelly devices in hard to reach places,
+// and you've disabled their Wireless Access Point for security reasons. Basically, during such scenarios,
+// you are locked out of the Shelly device's admin panel and can't connect to its AP because it's disabled
+// and you can't visit it's local IP, because it couldn't acquire an IP from the router. A simple restart
+// solves the problem in 99% of the cases.
+
+let CONFIG = {
+  // number of failures before triggering a Restart
+  numberOfFails: 5, // 5 minutes
+  // time in seconds between retries
+  retryIntervalSeconds: 60,
+};
+
+let failCounter = 0;
+let pingTimer = null;
+
+function checkForWifi() {
+  Shelly.call("WiFi.GetStatus",{}, function (response) {
```

Comment:
You can replace this with a synchronous call Shelly.getComponentStatus. It's more robust.

Reply:
@Mircho good suggestion. I didn't know about this utility function. Refactored.